cs50/2017/fall/recover/check50/__init__.py

```diff
@@ -72,25 +49,57 @@ def test_noimage(self):
     @check("compiles")
     def first_image(self):
         """recovers 000.jpg correctly"""
+        self.initialize_data()
```

Comment:
No need to call this in every check I don't think. Just move to constructor. Though just to be sure, @brianyu28, @crossroads1112, do we re-use the same `Check` child object?

Reply:
I did try doing it just once in a constructor but it didn't seem to work. It's possible something else was wrong, though. Same story with leaving out the self.add() for the data files, though I'll experiment with it more.

Reply:
With the way unittest works, you can't preserve instance variables across checks. `self.add` should propagate to all dependent checks though (e.g. it's pretty common to  `self.add` something most of the checks use in `exists` so that you don't have to call `self.add` in each of them).

Reply:
`initialize_data` should not be called in every check though. Probably best to move the `self.add` calls to the `exists` check and put the rest in a class method that's called at the end of the file.